api/applications_api.py

```diff
@@ -140,22 +140,24 @@ class ApplicationEventSerializer(serializers.ModelSerializer):
     )
 
     application_id = serializers.PrimaryKeyRelatedField(
-        queryset=Application.objects.all(), source="application"
```

Comment:
I think this application_id  should be required, i think it'll lead to pretty much these just disappearing when fetching an application with it's events. Creating these doesn't really make sense without the application existing.

Reply:
> I think this application_id should be required, i think it'll lead to pretty much these just disappearing when fetching an application with it's events. Creating these doesn't really make sense without the application existing.

Hmm, yeah, I was thinking this only from nested write perspective where application ID was set automatically. Wanted to make it possible to create an application with events in one POST call and in that case there is no application ID yet. But as you said it will probably cause some problems when using ApplicationEvent endpoint independently. Also probably there isn't a use case where whole application isn't already saved before saving application events. I'll change that.